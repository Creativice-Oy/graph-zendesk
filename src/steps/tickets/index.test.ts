import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import {
  buildTicketAssigneeRelationships,
  buildTicketGroupRelationships,
  buildTicketRequesterRelationships,
  fetchTickets,
} from '.';
import { integrationConfig } from '../../../test/config';
import { setupZendeskRecording } from '../../../test/recording';
import { IntegrationConfig } from '../../config';
import { Entities, Relationships } from '../constants';
import { fetchGroups } from '../groups';
import { fetchUsers } from '../users';

describe('#fetchTickets', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupZendeskRecording({
      directory: __dirname,
      name: 'fetchTickets',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchTickets(context);

    const tickets = context.jobState.collectedEntities.filter((e) =>
      e._type.includes(Entities.TICKET._type),
    );

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(tickets.length).toBeGreaterThan(0);
    expect(tickets).toMatchGraphObjectSchema({
      _class: Entities.TICKET._class,
      schema: {
        additionalProperties: false,
        required: [],
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: Entities.TICKET._type },
          _class: { const: Entities.TICKET._class },
          _key: { type: 'string' },
          active: { type: 'boolean' },
          assigneeEmail: { type: 'string' },
          assigneeId: { type: 'number' },
          brandId: { type: 'number' },
          collaboratorIds: { type: 'array', items: { type: 'number' } },
          createdAt: { type: 'string' },
          description: { type: 'string' },
          dueAt: { type: 'string' },
          emailCcIds: { type: 'array', items: { type: 'number' } },
          followerIds: { type: 'array', items: { type: 'number' } },
          forumTopicId: { type: 'number' },
          groupId: { type: 'number' },
          hasIncidents: { type: 'boolean' },
          id: { type: 'string' },
          name: { type: 'string' },
          organizationId: { type: 'number' },
          priority: { type: 'string' },
          problemId: { type: 'number' },
          recipient: { type: 'string' },
          requestedId: { type: 'number' },
          requesterId: { type: 'number' },
          status: { type: 'string' },
          submitterId: { type: 'number' },
          type: { type: 'string' },
          url: { type: 'string' },
          queries: { type: 'array', items: { type: 'string' } },
          tags: { type: 'array', items: { type: 'string' } },
        },
      },
    });
  });
});

describe('#buildTicketRequesterRelationships', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupZendeskRecording({
      directory: __dirname,
      name: 'buildTicketRequesterRelationships',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchUsers(context);
    await fetchTickets(context);
    await buildTicketRequesterRelationships(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === Relationships.USER_OPENED_TICKET._type,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'OPENED' },
          _type: {
            const: 'zendesk_user_opened_ticket',
          },
        },
      },
    });
  });
});

describe('#buildTicketAssigneeRelationships', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupZendeskRecording({
      directory: __dirname,
      name: 'buildTicketAssigneeRelationships',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchUsers(context);
    await fetchTickets(context);
    await buildTicketAssigneeRelationships(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === Relationships.USER_ASSIGNED_TICKET._type,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'ASSIGNED' },
          _type: {
            const: 'zendesk_user_assigned_ticket',
          },
        },
      },
    });
  });
});

describe('#buildTicketGroupRelationships', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupZendeskRecording({
      directory: __dirname,
      name: 'buildTicketGroupRelationships',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchTickets(context);
    await fetchGroups(context);
    await buildTicketGroupRelationships(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === Relationships.GROUP_HAS_TICKET._type,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'HAS' },
          _type: {
            const: 'zendesk_group_has_ticket',
          },
        },
      },
    });
  });
});
