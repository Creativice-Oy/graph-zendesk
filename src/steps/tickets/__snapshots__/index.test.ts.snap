// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#buildTicketAssigneeRelationships should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "zendesk_user:420755677092",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "active": true,
            "alias": null,
            "created_at": "2021-10-10T20:49:22Z",
            "custom_role_id": null,
            "default_group_id": 360021738231,
            "details": null,
            "email": "development@jupiterone.dev",
            "external_id": null,
            "iana_time_zone": "America/New_York",
            "id": 420755677092,
            "last_login_at": "2021-11-16T16:05:21Z",
            "locale": "en-US",
            "locale_id": 1,
            "moderator": true,
            "name": "Austin Kelleher",
            "notes": null,
            "only_private_comments": false,
            "organization_id": 361584776731,
            "phone": null,
            "photo": null,
            "report_csv": true,
            "restricted_agent": false,
            "role": "admin",
            "role_type": null,
            "shared": false,
            "shared_agent": false,
            "shared_phone_number": null,
            "signature": null,
            "suspended": false,
            "tags": Array [],
            "ticket_restriction": null,
            "time_zone": "America/New_York",
            "two_factor_auth_enabled": null,
            "updated_at": "2021-11-16T16:05:21Z",
            "url": "https://jupiterone-dev.zendesk.com/api/v2/users/420755677092.json",
            "user_fields": Object {},
            "verified": true,
          },
        },
      ],
      "_type": "zendesk_user",
      "active": true,
      "chatOnly": undefined,
      "createdAt": "2021-10-10T20:49:22Z",
      "createdOn": undefined,
      "defaultGroupId": 360021738231,
      "displayName": "Austin Kelleher",
      "email": "development@jupiterone.dev",
      "ianaTimeZone": "America/New_York",
      "id": "420755677092",
      "locale": "en-US",
      "localeId": 1,
      "moderator": true,
      "name": "Austin Kelleher",
      "organizationId": 361584776731,
      "phone": undefined,
      "photoUrl": undefined,
      "restrictedAgent": false,
      "role": "admin",
      "shared": false,
      "suspended": false,
      "timeZone": "America/New_York",
      "updatedAt": "2021-11-16T16:05:21Z",
      "url": "https://jupiterone-dev.zendesk.com/api/v2/users/420755677092.json",
      "username": "development@jupiterone.dev",
      "verified": true,
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "zendesk_user:420924998451",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "active": true,
            "alias": null,
            "created_at": "2021-10-10T20:49:34Z",
            "custom_role_id": null,
            "default_group_id": null,
            "details": null,
            "email": "customer@example.com",
            "external_id": null,
            "iana_time_zone": "America/New_York",
            "id": 420924998451,
            "last_login_at": null,
            "locale": "en-US",
            "locale_id": 1,
            "moderator": false,
            "name": "The Customer",
            "notes": null,
            "only_private_comments": false,
            "organization_id": null,
            "phone": null,
            "photo": Object {
              "content_type": "image/png",
              "content_url": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198.png",
              "deleted": false,
              "file_name": "profile_image_420924998451_11522198.png",
              "height": 80,
              "id": 361209460612,
              "inline": false,
              "mapped_content_url": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198.png",
              "size": 10254,
              "thumbnails": Array [
                Object {
                  "content_type": "image/png",
                  "content_url": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198_thumb.png",
                  "deleted": false,
                  "file_name": "profile_image_420924998451_11522198_thumb.png",
                  "height": 32,
                  "id": 361209460632,
                  "inline": false,
                  "mapped_content_url": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198_thumb.png",
                  "size": 2402,
                  "url": "https://jupiterone-dev.zendesk.com/api/v2/attachments/361209460632.json",
                  "width": 32,
                },
              ],
              "url": "https://jupiterone-dev.zendesk.com/api/v2/attachments/361209460612.json",
              "width": 80,
            },
            "report_csv": false,
            "restricted_agent": true,
            "role": "end-user",
            "role_type": null,
            "shared": false,
            "shared_agent": false,
            "shared_phone_number": null,
            "signature": null,
            "suspended": false,
            "tags": Array [],
            "ticket_restriction": "requested",
            "time_zone": "America/New_York",
            "two_factor_auth_enabled": false,
            "updated_at": "2021-10-10T20:49:36Z",
            "url": "https://jupiterone-dev.zendesk.com/api/v2/users/420924998451.json",
            "user_fields": Object {},
            "verified": false,
          },
        },
      ],
      "_type": "zendesk_user",
      "active": true,
      "chatOnly": undefined,
      "createdAt": "2021-10-10T20:49:34Z",
      "createdOn": undefined,
      "defaultGroupId": undefined,
      "displayName": "The Customer",
      "email": "customer@example.com",
      "ianaTimeZone": "America/New_York",
      "id": "420924998451",
      "locale": "en-US",
      "localeId": 1,
      "moderator": false,
      "name": "The Customer",
      "organizationId": undefined,
      "phone": undefined,
      "photoUrl": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198.png",
      "restrictedAgent": true,
      "role": "end-user",
      "shared": false,
      "suspended": false,
      "timeZone": "America/New_York",
      "updatedAt": "2021-10-10T20:49:36Z",
      "url": "https://jupiterone-dev.zendesk.com/api/v2/users/420924998451.json",
      "username": "customer@example.com",
      "verified": false,
    },
    Object {
      "_class": Array [
        "Question",
      ],
      "_key": "zendesk_ticket:1",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "allow_attachments": true,
            "allow_channelback": false,
            "assignee_id": 420755677092,
            "brand_id": 360006992911,
            "collaborator_ids": Array [],
            "created_at": "2021-10-10T20:49:34Z",
            "custom_fields": Array [],
            "description": "Hi there,

I’m sending an email because I’m having a problem setting up your new product. Can you help me troubleshoot?

Thanks,
 The Customer

",
            "due_at": null,
            "email_cc_ids": Array [],
            "external_id": null,
            "fields": Array [],
            "follower_ids": Array [],
            "followup_ids": Array [],
            "forum_topic_id": null,
            "group_id": 360021738231,
            "has_incidents": false,
            "id": 1,
            "is_public": true,
            "organization_id": null,
            "priority": "normal",
            "problem_id": null,
            "raw_subject": "Sample ticket: Meet the ticket",
            "recipient": null,
            "requester_id": 420924998451,
            "satisfaction_rating": null,
            "sharing_agreement_ids": Array [],
            "status": "open",
            "subject": "Sample ticket: Meet the ticket",
            "submitter_id": 420755677092,
            "tags": Array [
              "sample",
              "support",
              "zendesk",
            ],
            "type": "incident",
            "updated_at": "2021-10-10T20:49:35Z",
            "url": "https://jupiterone-dev.zendesk.com/api/v2/tickets/1.json",
            "via": Object {
              "channel": "sample_ticket",
              "source": Object {
                "from": Object {},
                "rel": null,
                "to": Object {},
              },
            },
          },
        },
      ],
      "_type": "zendesk_ticket",
      "active": undefined,
      "assigneeEmail": undefined,
      "assigneeId": 420755677092,
      "brandId": 360006992911,
      "collaboratorIds": Array [],
      "createdAt": "2021-10-10T20:49:34Z",
      "createdOn": undefined,
      "description": "Hi there,

I’m sending an email because I’m having a problem setting up your new product. Can you help me troubleshoot?

Thanks,
 The Customer

",
      "displayName": "Sample ticket: Meet the ticket",
      "dueAt": undefined,
      "emailCcIds": Array [],
      "followerIds": Array [],
      "forumTopicId": undefined,
      "groupId": 360021738231,
      "hasIncidents": false,
      "id": "1",
      "name": "Sample ticket: Meet the ticket",
      "organizationId": undefined,
      "priority": "normal",
      "problemId": undefined,
      "queries": Array [
        "Hi there,

I’m sending an email because I’m having a problem setting up your new product. Can you help me troubleshoot?

Thanks,
 The Customer

",
      ],
      "recipient": undefined,
      "requestedId": undefined,
      "requesterId": 420924998451,
      "status": "open",
      "submitterId": 420755677092,
      "tags": Array [
        "sample",
        "support",
        "zendesk",
      ],
      "type": "incident",
      "url": "https://jupiterone-dev.zendesk.com/api/v2/tickets/1.json",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "ASSIGNED",
      "_fromEntityKey": "zendesk_user:420755677092",
      "_key": "zendesk_user:420755677092|assigned|zendesk_ticket:1",
      "_toEntityKey": "zendesk_ticket:1",
      "_type": "zendesk_user_assigned_ticket",
      "displayName": "ASSIGNED",
    },
  ],
  "encounteredTypes": Array [
    "zendesk_user",
    "zendesk_ticket",
    "zendesk_user_assigned_ticket",
  ],
  "numCollectedEntities": 3,
  "numCollectedRelationships": 1,
}
`;

exports[`#buildTicketGroupRelationships should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Question",
      ],
      "_key": "zendesk_ticket:1",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "allow_attachments": true,
            "allow_channelback": false,
            "assignee_id": 420755677092,
            "brand_id": 360006992911,
            "collaborator_ids": Array [],
            "created_at": "2021-10-10T20:49:34Z",
            "custom_fields": Array [],
            "description": "Hi there,

I’m sending an email because I’m having a problem setting up your new product. Can you help me troubleshoot?

Thanks,
 The Customer

",
            "due_at": null,
            "email_cc_ids": Array [],
            "external_id": null,
            "fields": Array [],
            "follower_ids": Array [],
            "followup_ids": Array [],
            "forum_topic_id": null,
            "group_id": 360021738231,
            "has_incidents": false,
            "id": 1,
            "is_public": true,
            "organization_id": null,
            "priority": "normal",
            "problem_id": null,
            "raw_subject": "Sample ticket: Meet the ticket",
            "recipient": null,
            "requester_id": 420924998451,
            "satisfaction_rating": null,
            "sharing_agreement_ids": Array [],
            "status": "open",
            "subject": "Sample ticket: Meet the ticket",
            "submitter_id": 420755677092,
            "tags": Array [
              "sample",
              "support",
              "zendesk",
            ],
            "type": "incident",
            "updated_at": "2021-10-10T20:49:35Z",
            "url": "https://jupiterone-dev.zendesk.com/api/v2/tickets/1.json",
            "via": Object {
              "channel": "sample_ticket",
              "source": Object {
                "from": Object {},
                "rel": null,
                "to": Object {},
              },
            },
          },
        },
      ],
      "_type": "zendesk_ticket",
      "active": undefined,
      "assigneeEmail": undefined,
      "assigneeId": 420755677092,
      "brandId": 360006992911,
      "collaboratorIds": Array [],
      "createdAt": "2021-10-10T20:49:34Z",
      "createdOn": undefined,
      "description": "Hi there,

I’m sending an email because I’m having a problem setting up your new product. Can you help me troubleshoot?

Thanks,
 The Customer

",
      "displayName": "Sample ticket: Meet the ticket",
      "dueAt": undefined,
      "emailCcIds": Array [],
      "followerIds": Array [],
      "forumTopicId": undefined,
      "groupId": 360021738231,
      "hasIncidents": false,
      "id": "1",
      "name": "Sample ticket: Meet the ticket",
      "organizationId": undefined,
      "priority": "normal",
      "problemId": undefined,
      "queries": Array [
        "Hi there,

I’m sending an email because I’m having a problem setting up your new product. Can you help me troubleshoot?

Thanks,
 The Customer

",
      ],
      "recipient": undefined,
      "requestedId": undefined,
      "requesterId": 420924998451,
      "status": "open",
      "submitterId": 420755677092,
      "tags": Array [
        "sample",
        "support",
        "zendesk",
      ],
      "type": "incident",
      "url": "https://jupiterone-dev.zendesk.com/api/v2/tickets/1.json",
    },
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "zendesk_group:360021738231",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "created_at": "2021-10-10T20:49:30Z",
            "default": true,
            "deleted": false,
            "description": "",
            "id": 360021738231,
            "name": "Support",
            "updated_at": "2021-10-10T20:49:30Z",
            "url": "https://jupiterone-dev.zendesk.com/api/v2/groups/360021738231.json",
          },
        },
      ],
      "_type": "zendesk_group",
      "createdAt": "2021-10-10T20:49:30Z",
      "createdOn": undefined,
      "deleted": false,
      "description": "",
      "displayName": "Support",
      "id": "360021738231",
      "name": "Support",
      "updatedAt": "2021-10-10T20:49:30Z",
      "url": "https://jupiterone-dev.zendesk.com/api/v2/groups/360021738231.json",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "zendesk_group:360021738231",
      "_key": "zendesk_group:360021738231|has|zendesk_ticket:1",
      "_toEntityKey": "zendesk_ticket:1",
      "_type": "zendesk_group_has_ticket",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": Array [
    "zendesk_ticket",
    "zendesk_group",
    "zendesk_group_has_ticket",
  ],
  "numCollectedEntities": 2,
  "numCollectedRelationships": 1,
}
`;

exports[`#buildTicketRequesterRelationships should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "zendesk_user:420755677092",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "active": true,
            "alias": null,
            "created_at": "2021-10-10T20:49:22Z",
            "custom_role_id": null,
            "default_group_id": 360021738231,
            "details": null,
            "email": "development@jupiterone.dev",
            "external_id": null,
            "iana_time_zone": "America/New_York",
            "id": 420755677092,
            "last_login_at": "2021-11-16T16:05:21Z",
            "locale": "en-US",
            "locale_id": 1,
            "moderator": true,
            "name": "Austin Kelleher",
            "notes": null,
            "only_private_comments": false,
            "organization_id": 361584776731,
            "phone": null,
            "photo": null,
            "report_csv": true,
            "restricted_agent": false,
            "role": "admin",
            "role_type": null,
            "shared": false,
            "shared_agent": false,
            "shared_phone_number": null,
            "signature": null,
            "suspended": false,
            "tags": Array [],
            "ticket_restriction": null,
            "time_zone": "America/New_York",
            "two_factor_auth_enabled": null,
            "updated_at": "2021-11-16T16:05:21Z",
            "url": "https://jupiterone-dev.zendesk.com/api/v2/users/420755677092.json",
            "user_fields": Object {},
            "verified": true,
          },
        },
      ],
      "_type": "zendesk_user",
      "active": true,
      "chatOnly": undefined,
      "createdAt": "2021-10-10T20:49:22Z",
      "createdOn": undefined,
      "defaultGroupId": 360021738231,
      "displayName": "Austin Kelleher",
      "email": "development@jupiterone.dev",
      "ianaTimeZone": "America/New_York",
      "id": "420755677092",
      "locale": "en-US",
      "localeId": 1,
      "moderator": true,
      "name": "Austin Kelleher",
      "organizationId": 361584776731,
      "phone": undefined,
      "photoUrl": undefined,
      "restrictedAgent": false,
      "role": "admin",
      "shared": false,
      "suspended": false,
      "timeZone": "America/New_York",
      "updatedAt": "2021-11-16T16:05:21Z",
      "url": "https://jupiterone-dev.zendesk.com/api/v2/users/420755677092.json",
      "username": "development@jupiterone.dev",
      "verified": true,
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "zendesk_user:420924998451",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "active": true,
            "alias": null,
            "created_at": "2021-10-10T20:49:34Z",
            "custom_role_id": null,
            "default_group_id": null,
            "details": null,
            "email": "customer@example.com",
            "external_id": null,
            "iana_time_zone": "America/New_York",
            "id": 420924998451,
            "last_login_at": null,
            "locale": "en-US",
            "locale_id": 1,
            "moderator": false,
            "name": "The Customer",
            "notes": null,
            "only_private_comments": false,
            "organization_id": null,
            "phone": null,
            "photo": Object {
              "content_type": "image/png",
              "content_url": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198.png",
              "deleted": false,
              "file_name": "profile_image_420924998451_11522198.png",
              "height": 80,
              "id": 361209460612,
              "inline": false,
              "mapped_content_url": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198.png",
              "size": 10254,
              "thumbnails": Array [
                Object {
                  "content_type": "image/png",
                  "content_url": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198_thumb.png",
                  "deleted": false,
                  "file_name": "profile_image_420924998451_11522198_thumb.png",
                  "height": 32,
                  "id": 361209460632,
                  "inline": false,
                  "mapped_content_url": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198_thumb.png",
                  "size": 2402,
                  "url": "https://jupiterone-dev.zendesk.com/api/v2/attachments/361209460632.json",
                  "width": 32,
                },
              ],
              "url": "https://jupiterone-dev.zendesk.com/api/v2/attachments/361209460612.json",
              "width": 80,
            },
            "report_csv": false,
            "restricted_agent": true,
            "role": "end-user",
            "role_type": null,
            "shared": false,
            "shared_agent": false,
            "shared_phone_number": null,
            "signature": null,
            "suspended": false,
            "tags": Array [],
            "ticket_restriction": "requested",
            "time_zone": "America/New_York",
            "two_factor_auth_enabled": false,
            "updated_at": "2021-10-10T20:49:36Z",
            "url": "https://jupiterone-dev.zendesk.com/api/v2/users/420924998451.json",
            "user_fields": Object {},
            "verified": false,
          },
        },
      ],
      "_type": "zendesk_user",
      "active": true,
      "chatOnly": undefined,
      "createdAt": "2021-10-10T20:49:34Z",
      "createdOn": undefined,
      "defaultGroupId": undefined,
      "displayName": "The Customer",
      "email": "customer@example.com",
      "ianaTimeZone": "America/New_York",
      "id": "420924998451",
      "locale": "en-US",
      "localeId": 1,
      "moderator": false,
      "name": "The Customer",
      "organizationId": undefined,
      "phone": undefined,
      "photoUrl": "https://jupiterone-dev.zendesk.com/system/photos/361209460612/profile_image_420924998451_11522198.png",
      "restrictedAgent": true,
      "role": "end-user",
      "shared": false,
      "suspended": false,
      "timeZone": "America/New_York",
      "updatedAt": "2021-10-10T20:49:36Z",
      "url": "https://jupiterone-dev.zendesk.com/api/v2/users/420924998451.json",
      "username": "customer@example.com",
      "verified": false,
    },
    Object {
      "_class": Array [
        "Question",
      ],
      "_key": "zendesk_ticket:1",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "allow_attachments": true,
            "allow_channelback": false,
            "assignee_id": 420755677092,
            "brand_id": 360006992911,
            "collaborator_ids": Array [],
            "created_at": "2021-10-10T20:49:34Z",
            "custom_fields": Array [],
            "description": "Hi there,

I’m sending an email because I’m having a problem setting up your new product. Can you help me troubleshoot?

Thanks,
 The Customer

",
            "due_at": null,
            "email_cc_ids": Array [],
            "external_id": null,
            "fields": Array [],
            "follower_ids": Array [],
            "followup_ids": Array [],
            "forum_topic_id": null,
            "group_id": 360021738231,
            "has_incidents": false,
            "id": 1,
            "is_public": true,
            "organization_id": null,
            "priority": "normal",
            "problem_id": null,
            "raw_subject": "Sample ticket: Meet the ticket",
            "recipient": null,
            "requester_id": 420924998451,
            "satisfaction_rating": null,
            "sharing_agreement_ids": Array [],
            "status": "open",
            "subject": "Sample ticket: Meet the ticket",
            "submitter_id": 420755677092,
            "tags": Array [
              "sample",
              "support",
              "zendesk",
            ],
            "type": "incident",
            "updated_at": "2021-10-10T20:49:35Z",
            "url": "https://jupiterone-dev.zendesk.com/api/v2/tickets/1.json",
            "via": Object {
              "channel": "sample_ticket",
              "source": Object {
                "from": Object {},
                "rel": null,
                "to": Object {},
              },
            },
          },
        },
      ],
      "_type": "zendesk_ticket",
      "active": undefined,
      "assigneeEmail": undefined,
      "assigneeId": 420755677092,
      "brandId": 360006992911,
      "collaboratorIds": Array [],
      "createdAt": "2021-10-10T20:49:34Z",
      "createdOn": undefined,
      "description": "Hi there,

I’m sending an email because I’m having a problem setting up your new product. Can you help me troubleshoot?

Thanks,
 The Customer

",
      "displayName": "Sample ticket: Meet the ticket",
      "dueAt": undefined,
      "emailCcIds": Array [],
      "followerIds": Array [],
      "forumTopicId": undefined,
      "groupId": 360021738231,
      "hasIncidents": false,
      "id": "1",
      "name": "Sample ticket: Meet the ticket",
      "organizationId": undefined,
      "priority": "normal",
      "problemId": undefined,
      "queries": Array [
        "Hi there,

I’m sending an email because I’m having a problem setting up your new product. Can you help me troubleshoot?

Thanks,
 The Customer

",
      ],
      "recipient": undefined,
      "requestedId": undefined,
      "requesterId": 420924998451,
      "status": "open",
      "submitterId": 420755677092,
      "tags": Array [
        "sample",
        "support",
        "zendesk",
      ],
      "type": "incident",
      "url": "https://jupiterone-dev.zendesk.com/api/v2/tickets/1.json",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "OPENED",
      "_fromEntityKey": "zendesk_user:420924998451",
      "_key": "zendesk_user:420924998451|opened|zendesk_ticket:1",
      "_toEntityKey": "zendesk_ticket:1",
      "_type": "zendesk_user_opened_ticket",
      "displayName": "OPENED",
    },
  ],
  "encounteredTypes": Array [
    "zendesk_user",
    "zendesk_ticket",
    "zendesk_user_opened_ticket",
  ],
  "numCollectedEntities": 3,
  "numCollectedRelationships": 1,
}
`;

exports[`#fetchTickets should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Question",
      ],
      "_key": "zendesk_ticket:1",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "allow_attachments": true,
            "allow_channelback": false,
            "assignee_id": 420755677092,
            "brand_id": 360006992911,
            "collaborator_ids": Array [],
            "created_at": "2021-10-10T20:49:34Z",
            "custom_fields": Array [],
            "description": "Hi there,

I’m sending an email because I’m having a problem setting up your new product. Can you help me troubleshoot?

Thanks,
 The Customer

",
            "due_at": null,
            "email_cc_ids": Array [],
            "external_id": null,
            "fields": Array [],
            "follower_ids": Array [],
            "followup_ids": Array [],
            "forum_topic_id": null,
            "group_id": 360021738231,
            "has_incidents": false,
            "id": 1,
            "is_public": true,
            "organization_id": null,
            "priority": "normal",
            "problem_id": null,
            "raw_subject": "Sample ticket: Meet the ticket",
            "recipient": null,
            "requester_id": 420924998451,
            "satisfaction_rating": null,
            "sharing_agreement_ids": Array [],
            "status": "open",
            "subject": "Sample ticket: Meet the ticket",
            "submitter_id": 420755677092,
            "tags": Array [
              "sample",
              "support",
              "zendesk",
            ],
            "type": "incident",
            "updated_at": "2021-10-10T20:49:35Z",
            "url": "https://jupiterone-dev.zendesk.com/api/v2/tickets/1.json",
            "via": Object {
              "channel": "sample_ticket",
              "source": Object {
                "from": Object {},
                "rel": null,
                "to": Object {},
              },
            },
          },
        },
      ],
      "_type": "zendesk_ticket",
      "active": undefined,
      "assigneeEmail": undefined,
      "assigneeId": 420755677092,
      "brandId": 360006992911,
      "collaboratorIds": Array [],
      "createdAt": "2021-10-10T20:49:34Z",
      "createdOn": undefined,
      "description": "Hi there,

I’m sending an email because I’m having a problem setting up your new product. Can you help me troubleshoot?

Thanks,
 The Customer

",
      "displayName": "Sample ticket: Meet the ticket",
      "dueAt": undefined,
      "emailCcIds": Array [],
      "followerIds": Array [],
      "forumTopicId": undefined,
      "groupId": 360021738231,
      "hasIncidents": false,
      "id": "1",
      "name": "Sample ticket: Meet the ticket",
      "organizationId": undefined,
      "priority": "normal",
      "problemId": undefined,
      "queries": Array [
        "Hi there,

I’m sending an email because I’m having a problem setting up your new product. Can you help me troubleshoot?

Thanks,
 The Customer

",
      ],
      "recipient": undefined,
      "requestedId": undefined,
      "requesterId": 420924998451,
      "status": "open",
      "submitterId": 420755677092,
      "tags": Array [
        "sample",
        "support",
        "zendesk",
      ],
      "type": "incident",
      "url": "https://jupiterone-dev.zendesk.com/api/v2/tickets/1.json",
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "zendesk_ticket",
  ],
  "numCollectedEntities": 1,
  "numCollectedRelationships": 0,
}
`;
